File: Arrays/best_time_to_buy_and_sell_stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        price_left = 0
        price_right = 1
        max_profit = 0
        while price_right < len(prices):
            current_profit = prices[price_right] - prices[price_left]
            if prices[price_right] > prices[price_left]:
                max_profit = max(current_profit, max_profit)
            else:
                price_left = price_right
            price_right += 1
        return max_profit

File: Arrays/best_time_to_buy_and_sell_stock.ts
// https://leetcode.com/problems/best-time-to-buy-and-sell-stock/

function maxProfit(prices: number[]): number {
    let maxProfit: number = 0;
    let left: number = 0;
    let right: number = 1;
    while(right < prices.length){
        let currProfit: number = prices[right] - prices[left];
        maxProfit = Math.max(currProfit, maxProfit);
        if(prices[right] < prices[left]) left = right;
        right++;
    }
    return maxProfit;
};
